### ISSUE TITLE
Expand use of data from name references

### ISSUE THREAD

**Issue by dhobern** [CatalogueOfLife/portal-components#28]
The COLDP files I submit to ChecklistBank include page numbers for most names and direct page links (to BHL pages) for many of them.

See for example how I have rendered _Agdistopis sinhala_ here:

https://pterophoroidea.hobern.net/catalogue.php#taxon-12

The same data is presented in CLB like this:

https://www.checklistbank.org/dataset/1199/taxon/12

And in COL.org like this:

https://www.catalogueoflife.org/data/taxon/5TQCK

CLB shows more detail than COL.org - it would be nice to include all this in COL.org.

However, neither site shows the page numbers as provided in the source data and the links to the reference are the URLs from the reference record rather than the more specific ones in the name record.

**Comment by mdoering** [comment]
Yes indeed. There are several problems and inconsistencies how we render references. Showing links for the main publishedIn reference was an addressed issue before:
https://github.com/CatalogueOfLife/checklistbank/issues/1013

I would also prefer the links not to render the full URL but instead use an icon or small text like you do, Donald:

<img width="538" alt="image" src="https://user-images.githubusercontent.com/327505/177719020-e0ac7fc0-f76d-4666-8ea9-e8d9bbbb0fed.png">

The reference citation can be generated with html markup and without in the backend. It seems we expose and use the text one only, as otherwise the journal would be standing out in italics which would be nicer. I would suggest to treat references the same way as we do with names and provide both a `label` and `labelHtml` instead of the current `citation` field. @thomasstjerne or do we still render CSL client side with the citeproc library?

We show references in many places, including the reference details page:
https://www.checklistbank.org/dataset/1199/reference/601

The main issue you raised though is about additional infos that allow to refer to a single page for names and also for [taxon.accordingTo in the future](https://github.com/CatalogueOfLife/coldp/issues/60).
This is page and pageLink specifically. 

Here is a question: Do we want to deal with these 2 additionally to a reference which might already contain a page range, a DOI and a URL? Or would we rather modify the reference instance so the final CSL use just the more specific page and URL? I feel the later is better as it otherwise might be rather confusing to have both. And without it being part of the CSL instance we cannot include the specific page & link in the rendered citation but have to do this clientside somehow, e.g. in brackets behind the regular citation.

The specific page pointers only exist for Name.publishedIn and soon for Taxon.accordingTo. All other references don't have this option.

**Comment by mdoering** [comment]
I could also keep citation for some time while also adding label and labelHtml as the recommended thing to be more consistent?
